src/txmempool.cpp

```diff
@@ -16,6 +16,12 @@
 #include <util/system.h>
 #include <util/moneystr.h>
 #include <util/time.h>
+#include <validationinterface.h>
+
+// This map has to a separate global instead of a member of MainSignalsInstance,
+// because RegisterWithMempoolSignals is currently called before RegisterBackgroundSignalScheduler,
+// so MainSignalsInstance hasn't been created yet.
+static std::unordered_map<CTxMemPool*, boost::signals2::scoped_connection> g_connNotifyEntryRemoved;
```

Comment:
Why is this not a CTxMemPool member?

Reply:
Actually can be removed in one go:

07c8420f4ddddde7103d8a564f918389357904c9

Reply:
Thanks @MarcoFalke! Really appreciate your feedback. Please see 04390b2.

Reply:
nit: Does it makes sense to remove ` CTxMemPool::EntryRemoved`'s method and move its implementation to `CTxMemPool::removeUnchecked` ?

```
void CTxMemPool::removeUnchecked(txiter it, MemPoolRemovalReason reason)
{
    NotifyEntryRemoved(ptx, reason);
    if (reason != MemPoolRemovalReason::BLOCK && reason != MemPoolRemovalReason::CONFLICT) {
        GetMainSignals().TransactionRemovedFromMempool(ptx);
    }
    ...
}
```

Reply:
Sure, if you wish so